ext/sockets/sockets.c

```diff
@@ -2004,6 +2013,30 @@ PHP_FUNCTION(socket_set_option)
 		}
 #endif
 
+#ifdef SO_ATTACH_REUSEPORT_CBPF
+		case SO_ATTACH_REUSEPORT_CBPF: {
+			convert_to_long(arg4);
+			if (!Z_LVAL_P(arg4)) {
+				ov = 1;
+				optlen = sizeof(ov);
+				opt_ptr = &ov;
+				optname = SO_DETACH_BPF;
+			} else {
+				static struct sock_filter cbpf[] = {
+					BPF_STMT((BPF_LD|BPF_W|BPF_ABS), (uint32_t)(SKF_AD_OFF+SKF_AD_CPU)),
+					BPF_STMT((BPF_RET|BPF_A), 0),
```

Comment:
This kind support just one use case so it's sort of limited. I think it's useful but we are sort of introducing a single behavior that cannot be changed. Maybe we should use `arg4` as a constant that will identify type of the program that will be used. Then if someone comes with another classic BPF program use case it could be simply added. It would also allow us to overload arg4 in the future if we ever decide to provide some sort of interface to BPF.

Reply:
Yes I planned to pick it up at some point but too late for 8.2